tls/s2n_psk.c

```diff
@@ -599,3 +599,53 @@ int s2n_connection_set_psk_mode(struct s2n_connection *conn, s2n_psk_mode mode)
     conn->psk_mode_overridden = true;
     return S2N_SUCCESS;
 }
+
+int s2n_connection_get_chosen_psk(struct s2n_connection *conn, struct s2n_psk *chosen_psk)
+{
+    POSIX_ENSURE_REF(conn);
+    POSIX_ENSURE_REF(chosen_psk);
+
+    struct s2n_psk *conn_chosen_psk = conn->psk_params.chosen_psk;
+
+    if (conn_chosen_psk == NULL) {
+        return S2N_SUCCESS;
+    }
+
+    struct s2n_psk psk = *chosen_psk;
+    *chosen_psk = *conn_chosen_psk;
+    chosen_psk->identity = psk.identity;
+    POSIX_GUARD(s2n_psk_set_identity(chosen_psk, conn_chosen_psk->identity.data, conn_chosen_psk->identity.size));
```

Comment:
IDK about passing intentionally incomplete PSK objects to the customers. What if they took the result of this call and tried to append it to another connection? We can't have a s2n_psk structure that doesn't necessarily behave like an s2n_psk structure.

Should we not be wiping the chosen psk? Or should we have a separate structure for the chosen PSK?

Reply:
Maybe we should make this an s2n_offered_psk?